pygeos/test/test_creation_indices.py

```diff
@@ -1,36 +1,120 @@
 import pygeos
 import pytest
 import numpy as np
-from .common import point, line_string, linear_ring, polygon, empty
+from .common import point, line_string, linear_ring, polygon, empty, assert_geometries_equal
 
+pnts = pygeos.points
+lstrs = pygeos.linestrings
 geom_coll = pygeos.geometrycollections
 
 
 @pytest.mark.parametrize(
-    "geometries",
+    "func", [pygeos.points, pygeos.linestrings, pygeos.linearrings]
+)
+@pytest.mark.parametrize(
+    "coordinates",
     [
-        np.array([1, 2], dtype=np.int32),
+        np.empty((2,)),  # not enough dimensions
+        np.empty((2, 4, 1)),  # too many dimensions
+        np.empty((2, 4)),  # wrong inner dimension size
         None,
-        np.array([[point]]),
-        "hello",
+        np.full((2, 2), "foo", dtype=object),  # wrong type
     ],
 )
-def test_invalid_geometries(geometries):
-    with pytest.raises(TypeError):
-        pygeos.geometrycollections(geometries, indices=[0, 1])
+def test_invalid_coordinates(func, coordinates):
+    with pytest.raises((TypeError, ValueError)):
+        func(coordinates, indices=[0, 1])
+
+
+@pytest.mark.parametrize(
+    "func",
+    [
+        pygeos.multipoints,
+        pygeos.multilinestrings,
+        pygeos.multipolygons,
+        pygeos.geometrycollections,
+    ],
+)
+@pytest.mark.parametrize(
+    "geometries", [np.array([1, 2], dtype=np.int32), None, np.array([[point]]), "hello"]
+)
+def test_invalid_geometries(func, geometries):
+    with pytest.raises((TypeError, ValueError)):
+        func(geometries, indices=[0, 1])
+
+
+@pytest.mark.parametrize(
+    "func", [pygeos.points, pygeos.linestrings, pygeos.linearrings]
+)
+@pytest.mark.parametrize(
+    "indices", [np.array([point]), " hello", [0, 1]]  # wrong length
+)
+def test_invalid_indices_simple(func, indices):
+    with pytest.raises((TypeError, ValueError)):
+        func([[0.2, 0.3]], indices=indices)
+
+
+def test_points_invalid():
+    with pytest.raises(pygeos.GEOSException):
+        pygeos.points([[1, 1], [2, 2]], indices=[0, 0])
```

Comment:
It's not immediately obvious why this fails, perhaps adding either a comment or adding a `matches='Point coordinate list must contain a single element'` to `pytest.raises` would make that a little bit more obvious?